inbm/fpm/inb-provision-cloud/main.go

```diff
@@ -255,21 +257,54 @@ func configureUcc(cloudCredentialDir string, templateDir string) string {
 		jsonTemplate = createUnencryptedTemplate(templateDir)
 	}
 
-	clientId := getClientId()
+	clientId := getIdFromFile(uccClientIdFile)
+	if !isClientIdValid((clientId)) {
+		log.Fatalf("UCC Client ID does not meet the requirements.  Unable to provision for UCC.")
+	}
+
+	serverId := getIdFromFile(uccServerIdFile)
+	if !isServerIdValid(serverId) {
+		log.Fatalf("UCC Server ID does not meet the requirements.  Unable to provision for UCC.")
+	}
 
 	proxyHostName, proxyPort := configureProxy()
 	return makeCloudJson(ucc, jsonTemplate, caPath, deviceToken, serverIp, serverPort, deviceCertPath, deviceKeyPath,
-		proxyHostName, proxyPort, clientId)
+		proxyHostName, proxyPort, clientId, serverId)
 }
 
-func getClientId() string {
-	if content, err := ioutil.ReadFile(uccClientIdFile); err == nil {
+func getIdFromFile(filepath string) string {
+	if content, err := ioutil.ReadFile(filepath); err == nil {
 		return strings.TrimSpace(string(content))
 	}
-	log.Fatalf("Unable to read clientId.  Unable to provision for UCC")
+	log.Fatalf("Unable to read id from " + filepath + ".  Unable to provision for UCC.")
 	return ""
 }
 
+func isClientIdValid(id string) bool {
+	if len(id) == 0 || len(id) > 128 {
+		log.Fatalf("Client ID Length is greater than 128 characters.  Unable to provision for UCC.")
+		return false
+	}
+	if strings.ContainsAny(id, "# + \x00") {
+		return false
```

Comment:
Add a log here as well, else we wont know why it failed.
"Client ID contains invalid characters. Unable to provision for UCC."